### ISSUE TITLE
RichTextKit/SkiaSharp on Android: Could not resolve type with token 0100000b from typeref (expected class 'System.ReadOnlySpan`1' in assembly 'mscorlib')

### ISSUE THREAD

**Issue by mikegoatly** [dotnet/maui#5142]
### Description

When running a MAUI project on Android using RichTextKit to render text on a SkiaSharp `SKCanvasView` control the following exception is thrown:

```
System.TypeLoadException
  Message=Could not resolve type with token 0100000b from typeref (expected class 'System.ReadOnlySpan`1' in assembly 'mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e')
```

Repo here: https://github.com/mikegoatly/MauiExperiments/tree/main/MauiRichTextKit

I'm guessing that code is being trimmed that's needed?

### Steps to Reproduce

1. Create a new MAUI project
2. Add a package reference to `SkiaSharp.Views.Maui.Controls`
3. Add `.UseSkiaSharp()` to `MauiProgram`
4. Change `MainPage.xaml` to have this as content:
``` xaml
<Grid>
    <sk:SKCanvasView PaintSurface="SKCanvasView_PaintSurface"/>
</Grid>
```
5. Add this to `MainPage.xaml.cs`:
```  csharp
private void SKCanvasView_PaintSurface(object sender, SkiaSharp.Views.Maui.SKPaintSurfaceEventArgs e)
{
  var text = new TextBlock();
  var styleManager = StyleManager.Default.Value;
  styleManager.FontSize(40);
  styleManager.HaloColor(SKColors.Red);
  styleManager.HaloWidth(4);
  styleManager.HaloBlur(2);
  
  text.AddText("Hello MAUI!", styleManager.CurrentStyle);
  
  text.Paint(e.Surface.Canvas, new SKPoint(100, 200));
}
```

Debug the app on Android (I'm using `Pixel 5 - API 30 (Android 11.0 - API 30)`) and get the error. Note that the app works fine when running as a Windows app.

### Version with bug

Preview 13 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 11 and up

**Comment by Amir-Care** [comment]
Verified repro with Version 17.2.0 Preview 2.0 [32208.484.main] on Android.

- Throw exception:
   
  ![image](https://user-images.githubusercontent.com/99456025/157363919-c59cb495-aa01-47e0-969c-2940d230d577.png)

Sample project: [#5142.zip](https://github.com/dotnet/maui/files/8211305/5142.zip)

**Comment by mikegoatly** [comment]
@mattleibow I've just updated the [repo project](https://github.com/mikegoatly/MauiExperiments/tree/main/MauiRichTextKit) to use SkiaSharp.Views.Maui.Controls 2.88.0-preview.232 and still get the same error.

The sample code is already running on the latest available version of RichTextKit, and the callstack at the point of error is:

```
0x2B in Topten.RichTextKit.TextBlock.CreateFontRun at C:\Users\Brad\Projects\RichTextKit\Topten.RichTextKit\TextBlock.cs:1305,17
0x65 in Topten.RichTextKit.TextBlock.FlushUnshapedRuns at C:\Users\Brad\Projects\RichTextKit\Topten.RichTextKit\TextBlock.cs:1236,17
0x367 in Topten.RichTextKit.TextBlock.BuildFontRuns at C:\Users\Brad\Projects\RichTextKit\Topten.RichTextKit\TextBlock.cs:1129,17
0xF6 in Topten.RichTextKit.TextBlock.Layout at C:\Users\Brad\Projects\RichTextKit\Topten.RichTextKit\TextBlock.cs:313,17
0xB in Topten.RichTextKit.TextBlock.Paint at C:\Users\Brad\Projects\RichTextKit\Topten.RichTextKit\TextBlock.cs:359,13
0x1E in Topten.RichTextKit.TextBlock.Paint at C:\Users\Brad\Projects\RichTextKit\Topten.RichTextKit\TextBlock.cs:419,13
0x76 in MauiRichTextKit.MainPage.SKCanvasView_PaintSurface at C:\Users\mikeg\source\repos\MauiExperiments\MauiRichTextKit\MauiRichTextKit\MainPage.xaml.cs:25,2
0xD in SkiaSharp.Views.Maui.Controls.SKCanvasView.OnPaintSurface at D:\a\1\s\source\SkiaSharp.Views.Forms\SkiaSharp.Views.Forms.Shared\SKCanvasView.cs:74,4
0x2 in SkiaSharp.Views.Maui.Controls.SKCanvasView.SkiaSharp.Views.Maui.ISKCanvasView.OnPaintSurface at D:\a\1\s\source\SkiaSharp.Views.Maui\SkiaSharp.Views.Maui.Controls\SKCanvasView.HandlerImpl.cs:33,4
0x52 in SkiaSharp.Views.Maui.Handlers.SKCanvasViewHandler.OnPaintSurface at D:\a\1\s\source\SkiaSharp.Views.Maui\SkiaSharp.Views.Maui.Core\Handlers\SKCanvasView\SKCanvasViewHandler.Android.cs:68,4
0xD in SkiaSharp.Views.Android.SKCanvasView.OnPaintSurface at D:\a\1\s\source\SkiaSharp.Views\SkiaSharp.Views.Android\SKCanvasView.cs:134,4
0xB6 in SkiaSharp.Views.Android.SKCanvasView.OnDraw at D:\a\1\s\source\SkiaSharp.Views\SkiaSharp.Views.Android\SKCanvasView.cs:113,4
0x11 in Android.Views.View.n_OnDraw_Landroid_graphics_Canvas_ at /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net6.0/android-32/mcw/Android.Views.View.cs:18211,4
0x9 in Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPL_V at /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:110,5
```

Which is [this line](https://github.com/toptensoftware/RichTextKit/blob/5f4d2c80cc3723a7d3018172803c64637f92f7ee/Topten.RichTextKit/TextBlock.cs#L1305):

``` csharp
shaped = shaper.Shape(_textShapingBuffers, codePoints, style, direction, codePoints.Start, asFallbackFor, ResolveTextAlignment());
```

However because the error is a type resolution error, I don't think that line is actually to blame. Looking at what's going on in the `TextShaper.Shape` method that's being called, this is the first time I think the HarfBuzzSharp library will be invoked, so it's very possible that there's something being trimmed in *that* that's actually the problem here?

**Comment by softlion** [comment]
@mikegoatly have you resolved the issue ?  
I have the same issue with another lib. It may be an issue with the current maui toolset.

The current project hierarchy is:  
```nuget net7 (release) lib => debug net7 lib project => debug maui android .exe project```  
=> The runtime error when running the android app is the same as you, it does not find a type in the nuget package.

So I tried referencing directly the nuget lib from it's source code.  
```debug net7 lib project => debug net7 lib project => debug maui android .exe project```  
=> And surprise, I get the same error. That means it's not a release issue.
The error is, a type from the leftmost lib is not found by the middle lib when the exe instantiate an object having this type as member.

I tried on both VS windows and Rider windows.  
My conclusion is, the current maui toolset is not working as expected and not usable.

**Comment by sacOO7** [comment]
This is true, these people are planning to replace Xamarin with MAUI, yet MAUI is full of runtime bugs, mostly related to mscorlib.

**Comment by akoeplinger** [comment]
The reason for this error is because the build is pulling in an assembly that was compiled against monoandroid and that hits an issue with the compatibility shim, see my explanation here: https://github.com/dotnet/runtime/issues/82193#issuecomment-1442134342

The easiest solution is to add a `net6.0` target to whichever nuget is causing the issue. I'll see if there is some other way we can improve this for net8.0

**Comment by XamlTest** [comment]
Verified this issue on Visual Studio Enterprise 17.6.0 Preview 3.0. This issue does not repro on Android emulator (13.0-API 33) with below Project: [Uploading MauiRichTextKit.zip…]()
![image](https://user-images.githubusercontent.com/97647558/231696992-bf8143c9-b630-4956-9346-570880951b14.png)
